parachain/pallets/ethereum-beacon-client/fuzz/src/impls.rs

```diff
@@ -0,0 +1,191 @@
+            signature_slot: other.signature_slot,
+            next_sync_committee_update: next,
+            finalized_header: other.finalized_header.clone().try_into().unwrap(),
+            finality_branch: other.finality_branch.iter().map(|&hash| {
+                H256::from(hash)
+            }).collect::<Vec<_>>().as_slice().try_into().unwrap(),
+            block_roots_root: other.block_roots_root.into(),
+            block_roots_branch: other.block_roots_branch.iter().map(|&hash| {
+                H256::from(hash)
+            }).collect::<Vec<_>>().as_slice().try_into().unwrap(),
+        })
+    }
+}
+
+impl TryFrom<FuzzNextSyncCommitteeUpdate> for NextSyncCommitteeUpdate
+{
+    type Error = String;
+
+    fn try_from(other: FuzzNextSyncCommitteeUpdate) -> Result<Self, Self::Error> {
+        Ok(Self {
+            next_sync_committee: SyncCommittee{
+                pubkeys: other.next_sync_committee.pubkeys.iter().map(|&pk| {
+                    let p: PublicKey = pk.into();
+                    p
+                }).collect::<Vec<_>>().as_slice().try_into().unwrap(),
+                aggregate_pubkey: other.next_sync_committee.aggregate_pubkey.into(),
+            },
+            next_sync_committee_branch: other.next_sync_committee_branch.iter().map(|&hash| {
+                H256::from(hash)
+            }).collect::<Vec<_>>().as_slice().try_into().unwrap(),
+        })
+    }
+}
+
+impl TryFrom<FuzzSyncAggregate> for SyncAggregate
+{
+    type Error = String;
+
+    fn try_from(other: FuzzSyncAggregate) -> Result<Self, Self::Error> {
+        Ok(Self {
+            sync_committee_bits: other.sync_committee_bits.into(),
+            sync_committee_signature: other.sync_committee_signature.into(),
+        })
+    }
+}
+
+impl TryFrom<FuzzCheckpointUpdate> for CheckpointUpdate
+{
+    type Error = String;
+
+    fn try_from(other: FuzzCheckpointUpdate) -> Result<Self, Self::Error> {
+        Ok(Self {
+            header: other.header.clone().try_into().unwrap(),
+            current_sync_committee: other.current_sync_committee.try_into().unwrap(),
+            current_sync_committee_branch: other.current_sync_committee_branch.iter().map(|&hash| {
+                H256::from(hash)
+            }).collect::<Vec<_>>().as_slice().try_into().unwrap(),
+            validators_root: other.validators_root.into(),
+            block_roots_root: other.block_roots_root.into(),
+            block_roots_branch: other.block_roots_branch.iter().map(|&hash| {
+                H256::from(hash)
+            }).collect::<Vec<_>>().as_slice().try_into().unwrap(),
+        })
+    }
+}
+
+impl TryFrom<FuzzSyncCommittee> for SyncCommittee
+{
+    type Error = String;
+
+    fn try_from(other: FuzzSyncCommittee) -> Result<Self, Self::Error> {
+
+        Ok(Self{
+            pubkeys: other.pubkeys.iter().map(|&pk| {
+                let p: PublicKey = pk.into();
+                p
+            }).collect::<Vec<_>>().as_slice().try_into().unwrap(),
```

Comment:
We can shorten the `map` here:
```suggestion
            pubkeys: other.pubkeys.iter()
                .map(|&pk| { pk.into() })
                .collect::<Vec<_>>().as_slice().try_into().unwrap(),
```